File: Data Structures & Algorithms/search-insert-position/submission-0.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1
        res = len(nums)

        while l<=r:
            m = (l+r)>>1
            
            if nums[m] < target:
                l = m+1
            elif nums[m] > target:
                res =m
                r = m-1
            else:
                return m
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> previousCheck = new HashMap<>();

        for(int x = 0; x < nums.length; x++){
            int num = nums[x]; 
            int diff = target - num; 

            if(previousCheck.containsKey(diff)){
                return new int[] {previousCheck.get(diff), x};
            }

            previousCheck.put(num, x);
        }
        return new int[] {};
    }
}
